src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2634,15 +2652,27 @@ private void EmitInitObj(TypeSymbol type, bool used, CSharpSyntaxNode syntaxNode
             }
         }
 
+        private void EmitGetTypeFromHandle(BoundTypeOf boundTypeOf)
+        {
+            _builder.EmitOpCode(ILOpCode.Call, stackAdjustment: 0); //argument off, return value on
+            var getTypeMethod = boundTypeOf.GetTypeFromHandle;
+            Debug.Assert((object)getTypeMethod != null); // Should have been checked during binding
+            EmitSymbolToken(getTypeMethod, boundTypeOf.Syntax, null);
+        }
+
+        private void EmitTypeOfPrivateImplementationDetailsExpression(BoundTypeOfPrivateImplementationDetails boundTypeOfOperator)
+        {
+            _builder.EmitOpCode(ILOpCode.Ldtoken);
+            EmitTypeReferenceToken(_module.PrivateImplClass, boundTypeOfOperator.Syntax);
```

Comment:
It looks like GetPrivateImplClass should be used instead, PrivateImplClass doesn't allocate the type.